actions/config.js

```diff
@@ -0,0 +1,128 @@
+
+function getSharepointConfig(config) {
+    const sharepointConfig = config.sp.data[0];
+    // ${sharepointConfig.site} - MS Graph API Url with site pointers.
+    const baseURI = `${sharepointConfig.site}/drive/root:${sharepointConfig.rootFolders}`;
+    const fgBaseURI = `${sharepointConfig.site}/drive/root:${sharepointConfig.fgRootFolder}`;
+    return {
+        ...sharepointConfig,
+        clientApp: {
+            auth: {
+                clientId: sharepointConfig.clientId,
+                authority: sharepointConfig.authority,
+            },
+            cache: { cacheLocation: 'sessionStorage' },
+        },
+        shareUrl: sharepointConfig.shareurl,
+        fgShareUrl: sharepointConfig.fgShareUrl,
+        login: { redirectUri: '/tools/loc/spauth' },
+        api: {
+            url: GRAPH_API,
+            file: {
+                get: { baseURI, fgBaseURI },
+                download: { baseURI: `${sharepointConfig.site}/drive/items` },
+                upload: {
+                    baseURI,
+                    fgBaseURI,
+                    method: 'PUT',
+                },
+                delete: {
+                    baseURI,
+                    fgBaseURI,
+                    method: 'DELETE',
+                },
+                update: {
+                    baseURI,
+                    fgBaseURI,
+                    method: 'PATCH',
+                },
+                createUploadSession: {
+                    baseURI,
+                    fgBaseURI,
+                    method: 'POST',
+                    payload: { '@microsoft.graph.conflictBehavior': 'replace' },
+                },
+                copy: {
+                    baseURI,
+                    fgBaseURI,
+                    method: 'POST',
+                    payload: { '@microsoft.graph.conflictBehavior': 'replace' },
+                },
+            },
+            directory: {
+                create: {
+                    baseURI,
+                    fgBaseURI,
+                    method: 'PATCH',
+                    payload: { folder: {} },
+                },
+            },
+            excel: {
+                update: {
+                    baseURI,
+                    fgBaseURI,
+                    method: 'POST',
+                },
+            },
+            batch: { uri: `${GRAPH_API}/$batch` },
+        },
+    };
+}
+
+async function fetchConfigJson(configPath) {
+    const logger = getAioLogger();
+    const configResponse = await fetch(configPath);
+    if (!configResponse.ok) {
+        logger.error('Config not found');
+    }
+    return configResponse.json();
+}
+
+function getHelixAdminConfig() {
+    const adminServerURL = 'https://admin.hlx.page';
+    return {
+        api: {
+            status: { baseURI: `${adminServerURL}/status` },
+            preview: { baseURI: `${adminServerURL}/preview` },
+        },
+    };
+}
+
+async function getConfig(adminPageUri) {
```

Comment:
Instead of repeatedly making a network call to sharepoint to fetch the floodgate config JSON, let's do this (in a separate PR):
- move sensitive and generic properties (i.e. `site`, `clientId` and `authority`) out of `config.xslx` to github secrets
- define environment variables for these secrets in `app.config.yaml` for actions that require them
- inject required secrets during github CI/CD actions
- continue passing project-specific parameters from milo floodgate tool to floodgate aio APIs